tests/test_limiter.py

```diff
@@ -0,0 +1,26 @@
+import datetime
+from unittest.mock import MagicMock
+from riven.limiter import rate_limiter
+
+
+class TestLimiter:
+
+    def test_limter_one_second_limit(self):
+        called_times = []
+
+        @rate_limiter
+        def mock_function(num):
+            request = MagicMock()
+            request.from_cache = False
+            request.__repr__ = lambda _: f"Requesting: {num}"
+            called_times.append(datetime.datetime.now())
+            return request
+
+        for i in range(100):
+            mock_function(i)
+
+        start = 0
+        for i in range(1, len(called_times)):
+            if called_times[i] - called_times[start] >= datetime.timedelta(0, 1):
+                assert i - start - 1 <= 20
+                start = i
```

Comment:
Feel like this one might miss some cases - say we have [t0... t19] >= 1, but 17 of those requests [t3... t19] happen in the last 0.1 seconds. We jump start index to 19, then e.g. [t19... t38] >= 1 but 10 of those are in the first 0.3 seconds of this window - we end up with [t3... t29] in 0.5 seconds. Very edge case though.

tldr; I think sliding window of 1 second width might be more accurate

Reply:
agreed. sliding window is a bit fiddly though